src/wallet/wallet.cpp

```diff
@@ -2149,6 +2149,11 @@ void CWallet::AvailableCoins(interfaces::Chain::Lock& locked_chain, std::vector<
         }
 
         for (unsigned int i = 0; i < wtx.tx->vout.size(); i++) {
+            // Only consider selected coins if add_inputs is false
+            if (coinControl && !coinControl->m_add_inputs && !coinControl->IsSelected(COutPoint(entry.first, i))) {
```

Comment:
ultra nit, cache `IsSelected()` by adding above:

```cpp
const bool is_selected = coinControl && coinControl->IsSelected(COutPoint(entry.first, i));
```

then use this in L2160 below.